cmd/configure.go

```diff
@@ -8,56 +8,113 @@ import (
 	"github.com/spf13/viper"
 )
 
-var configColors []*string
-
-var defaultColorKeys []string
-
-var defaultColors = map[string]string{
-	"gray":  "8",
-	"red":   "1",
-	"green": "2",
+// colorCfg holds config data for a single color
+// the `configValue` field is meant to be populated during the `init()` call to
+// be used by flags
+type colorCfg struct {
+	colorName    string
+	defaultValue string
+	configValue  *string
 }
 
-// configureCmd represents the configure command
+// configureCmd represents the configure command which is a container for other
+// sub-commands (e.g., colors, submission base URL)
 var configureCmd = &cobra.Command{
 	Use:   "configure",
 	Short: "Change configuration of the CLI",
+}
+
+var colorCfgs = []colorCfg{
+	{colorName: "gray", defaultValue: "8"},
+	{colorName: "red", defaultValue: "1"},
+	{colorName: "green", defaultValue: "2"},
+}
+var resetColors bool
+
+// configureColorsCmd represents the `configure colors` command for changing
+// the colors of the text output
+var configureColorsCmd = &cobra.Command{
+	Use:   "colors",
+	Short: "Change the CLI text colors",
 	Run: func(cmd *cobra.Command, args []string) {
+		if resetColors {
+			for _, cfg := range colorCfgs {
+				key := "color." + cfg.colorName
+				defaultValue := cfg.defaultValue
+				viper.Set(key, defaultValue)
+			}
+
+			viper.WriteConfig()
+			return
+		}
+
 		showHelp := true
-		for i, color := range configColors {
-			key := "color." + defaultColorKeys[i]
-			if color != nil {
-				if *color == "" {
-					viper.Set(key, defaultColors[defaultColorKeys[i]])
-					fmt.Println("unset " + key)
-				} else {
-					if viper.GetString(key) == *color {
-						continue
-					}
-					viper.Set(key, *color)
-					style := lipgloss.NewStyle().Foreground(lipgloss.Color(*color))
-					fmt.Println("set " + style.Render(key) + "!")
-				}
-				showHelp = false
+		for _, cfg := range colorCfgs {
+			key := "color." + cfg.colorName
+			cfgValue := cfg.configValue
+
+			if *cfgValue == "" {
+				continue
+			}
+
+			showHelp = false
+			if viper.GetString(key) == *cfgValue {
+				continue
 			}
+
+			viper.Set(key, *cfgValue)
+			style := lipgloss.NewStyle().Foreground(lipgloss.Color(*cfgValue))
+			fmt.Println("set " + style.Render(key) + "!")
 		}
+
 		if showHelp {
 			cmd.Help()
 		} else {
 			viper.WriteConfig()
 		}
+	},
+}
+
+// configureBaseURL variables
+var resetSubmitBaseURL bool
+
+// configureBaseURLCmd represents the `configure base_url` command
+var configureBaseURLCmd = &cobra.Command{
+	Use:   "base_url",
+	Short: "Set the base URL for HTTP tests, overriding lesson defaults",
+	Args:  cobra.RangeArgs(0, 1),
+	Run: func(cmd *cobra.Command, args []string) {
+		if resetSubmitBaseURL {
+			viper.Set("submit_base_url", "")
+			viper.WriteConfig()
```

Comment:
Same here
```suggestion
			viper.Set("submit_base_url", "")
			viper.WriteConfig()
			fmt.Println("Reset base URL")
```

probably also want to check for an err from `.WriteConfig()`